docs/library/ubluetooth.rst

```diff
@@ -493,6 +514,64 @@ device name from the device information service).
     peripheral initiating the MTU exchange. NimBLE works for both roles.
 
 
+L2CAP connection-oriented-channels (CoC)
+----------------------------------------
+
+    This feature allows for socket-like data exchange between two BLE devices.
+    Once the devices are connected via GAP, either device can listen for the
+    other to connect.
+
+    **Note:** This is currently only supported when using the NimBLE stack on
+    STM32 and Unix (not ESP32). Only one L2CAP CoC may be active at a given time
+    (i.e. you cannot connect while listening).
+
+.. method:: BLE.l2cap_coc_listen(psm, mtu, /)
+
+    Start listening for incoming L2CAP CoC requests on the specified *psm*.
+
+    The *mtu* specified will be the maximum payload size that this device can
+    transmit (which may be different to the MTU used in the other direction).
+
+    When a remote device initiates a connection, the ``_IRQ_L2CAP_COC_ACCEPT``
+    event will be raised, followed by the ``_IRQ_L2CAP_COC_CONNECT`` event. Both
+    events include a "channel handle", and the connect event includes the local
+    and remote MTU.
+
+.. method:: BLE.l2cap_coc_connect(conn_handle, psm, mtu, /)
+
+    Connect to a listening peer on the specified *psm*.
+
+    The *mtu* specified will be the maximum payload size that this device can
+    transmit (which may be different to the MTU used in the other direction).
+
+    On successful connection, the the ``_IRQ_L2CAP_COC_CONNECT`` event will be
+    raised, including a "channel handle" and the local and remote MTU.
+
+.. method:: BLE.l2cap_coc_disconnect(chan_handle, /)
+
+    Disconnect an active L2CAP CoC with the specified *chan_handle*.
+
+.. method:: BLE.l2cap_coc_send(chan_handle, buf, /)
+
+    Send the specified *buf* (which must support the buffer protocol) on the
+    L2CAP CoC identified by *chan_handle*.
+
+    This will return ``True`` if the channel is now "stalled", which means that
+    :meth:`l2cap_coc_send <BLE.l2cap_coc_send>` must not be called again until
+    the ``_IRQ_L2CAP_COC_TX_UNSTALLED`` event is received.
+
+    The buffer must always be smaller than the MTU.
+
+.. method:: BLE.l2cap_coc_recvinto(chan_handle, buf, /)
+
+    Receive data from the specified *chan_handle* into the provided *buf* (which
+    must support the buffer protocol, e.g. bytearray or memoryview).
+
+    Returns the number of bytes read from the channel.
```

Comment:
If there were more bytes available in the channel than the size of the user `buf`, it doesn't discard remaining bytes, right?  If so then it does actually act more like a stream than a packet-oriented (eg UDP) connection.

Reply:
Yes, exactly.

I've updated the docs to clarify that the user should always keep calling recvinto to drain the buffer so that the remote end can send again.